src/commands/lottery.ts

```diff
@@ -0,0 +1,99 @@
+import { User } from '../models/user.model';
+import { Lottery } from '../models/lottery.model';
+import { MessageEmbed } from 'discord.js';
+import { colors, version } from '../config/config';
+import { currency } from '../utils/format';
+import { msgCooldown } from '../config/cooldowns';
+
+const entryFees = {
+    daily: 1000,
+    weekly: 5000,
+    monthly: 25000
+};
+
+// capitalize first letter
+const capitalize = (str: string) => str.charAt(0).toUpperCase() + str.slice(1);
+
+// calculate time between now and end date
+const subtractDate = endDate => {
+    const now = new Date();
+    let timeLeft = endDate.getTime() - now.getTime();
+
+    if (timeLeft < 0) {
+        timeLeft = 0;
+    }
+
+    return timeLeft;
+};
+
+const enterUser = async (client, msg, type, cost) => {
+    const lottery = await Lottery.findOne({ type });
+    const user = await User.findOne({ discordId: msg.author.id });
+
+    if (lottery.entries.includes(msg.author.id)) {
+        return msg.reply('You are already entered!');
+    }
+
+    if (user.balance < cost) {
+        return msg.reply('You do not have enough to enter this lottery.');
+    }
+
+    client.logger.log(`(${msg.author.id}) entered the ${type} lottery.`);
+    await User.updateOne({ discordId: msg.author.id }, {
+        balance: user.balance - cost
+    });
+
+    await Lottery.updateOne({ type }, {
+        entries: [...lottery.entries, msg.author.id]
+    });
+
+    return msg.reply('You have been entered! 🍀');
+};
+
+// format time to dd:hh:mm:ss
+const formatTime = ms => {
+    let seconds = parseInt((ms / 1000) % 60);
+    let minutes = parseInt((ms / (1000 * 60)) % 60);
+    let hours = parseInt((ms / (1000 * 60 * 60)) % 24);
+    const days = parseInt((ms / (1000 * 60 * 60 * 24)) % 365);
+
+    hours = (hours < 10) ? '0' + hours : hours;
+    minutes = (minutes < 10) ? '0' + minutes : minutes;
+    seconds = (seconds < 10) ? '0' + seconds : seconds;
+
+    return `${days ? `${days}:` : ''}${hours}:${minutes}:${seconds}`;
+};
+
+export default async (msg, client, args) => {
+    const lotteryTypes = ['daily', 'weekly', 'monthly'];
+    if (args[1] === 'enter' && lotteryTypes.includes(args[0])) {
```

Comment:
Consider the capitalization of the argument. (e.g. `$lottery Daily enter` or `$lottery DAILY enter`)